### DISCUSSION TITLE
Mojo syntax highlighting in Quarto

### DISCUSSION THREAD

**Discussion by fnands** [modular/modular#1583] [Q&A] [answered]
Hi, 

I'm moving my blog over to Quarto, and I know that's what's used for the Mojo docs, so how is syntax highlighting dealt with there? 

I was looking at [this post](https://hamel.dev/notes/quarto/highlighting.html), and I see Quarto uses the KDE XML syntax specifications. 

The [official Mojo highlighting spec](https://github.com/modularml/mojo-syntax) is done in Textmate format, so what is being done for the official docs? 

Is there a `.xml` version you can share? 

Pinging @scottamain as he seems to be in charge of the docs.

**Comment by scottamain** [comment, marked as answer]
Hi @fnands. Yeah, we have a modified version of [the Python syntax definition](https://github.com/KDE/syntax-highlighting/blob/master/data/syntax/python.xml) we use like this:
```
format:
  html:
    syntax-definitions: 
    - mojo.xml
```
I'd be happy to share our `mojo.xml` file, but I'll need a few days to do it properly. We won't be using Quarto for much longer, so I'd prefer to open a PR to merge our file in the [KDE syntax-highlighting repo](https://github.com/KDE/syntax-highlighting/tree/master/data/syntax) for posterity.

**Reply by fnands** [reply]
> so I'd prefer to open a PR to merge our file in the [KDE syntax-highlighting repo](https://github.com/KDE/syntax-highlighting/tree/master/data/syntax) for posterity.

Thanks, that seems like the best solution. 

Any specific reason you are moving away from Quarto?

**Reply by scottamain** [reply]
For more customization. Quarto has been great for many reasons, but it's more difficult to extend/modify with custom UI components compared to other options.

**Comment by Texl** [comment]
I'm also interested in this - I'll keep an eye out at the linked repo.

**Comment by scottamain** [comment]
Hi @fnands and @Texl, sorry for the delay, but it's going to take a bit longer to merge this properly because any new syntax file requires tests, and I just don't have time to do that right now. So in the meantime, I just want to enable you to build your Mojo docs by sharing [the current mojo.xml file with you here](https://github.com/scottamain/syntax-highlighting/commit/eb9af4dc1243c77d0c638b6436afcd0702f2bebf). I hope this helps! 🍻

**Reply by fnands** [reply]
Thanks!
